pkg/flow/internal/controller/loader.go

```diff
@@ -95,14 +94,23 @@ func NewLoader(globals ComponentGlobals) *Loader {
 // The provided parentContext can be used to provide global variables and
 // functions to components. A child context will be constructed from the parent
 // to expose values of other components.
-func (l *Loader) Apply(parentScope *vm.Scope, componentBlocks []*ast.BlockStmt, configBlocks []*ast.BlockStmt) diag.Diagnostics {
+func (l *Loader) Apply(args *map[string]any, componentBlocks []*ast.BlockStmt, configBlocks []*ast.BlockStmt) diag.Diagnostics {
 	start := time.Now()
 	l.mut.Lock()
 	defer l.mut.Unlock()
 	l.cm.controllerEvaluation.Set(1)
 	defer l.cm.controllerEvaluation.Set(0)
 
-	newGraph, diags := l.loadNewGraph(parentScope, componentBlocks, configBlocks)
+	if args != nil {
+		for key, value := range *args {
+			l.cache.CacheExports(
+				ComponentID{"argument", key, "value"},
+				value,
+			)
+		}
+	}
```

Comment:
I think this was covered by `l.cache.SyncIDs(componentIDs)` toward the end of Apply since I was reusing the existing exports cache. With the update to put these in their own cache that may change.